fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/JoinPredicatePushdown.java

```diff
@@ -403,38 +407,38 @@ private void getPushdownPredicatesFromEquivalenceDerive(
         if (join.getJoinType().isLeftSemiJoin()) {
             for (ScalarOperator p : derivedPredicates) {
                 if (rightOutputColumns.containsAll(p.getUsedColumns())) {
-                    p.setIsPushdown(true);
+                    optimizerContext.addPushdownDerivedPredicate(p);
```

Comment:
What's the difference between `p.setIsPushdown(true);` and `optimizerContext.addPushdownDerivedPredicate(p);`?

Reply:
optimizerContext.addPushdownDerivedPredicate(p) is global effective, but isPushdown may be lost.